### ISSUE TITLE
Are there any examples using this in an async/await scenario?

### ISSUE THREAD

**Issue by tillig** [alexandrnikitin/CircuitBreaker.Net#1]
I have a scenario where I'm trying to call a REST API service using `HttpClient` and use the circuit breaker for service timeouts.

I think I'm getting lost in the async/await/`TaskScheduler` thing and I'm finding the asynchronous operations timing out aren't triggering any `CircuitBreakerTimeoutException` nor is the circuit breaker ever opening.

My code in an async/await method looks like this:

``` csharp
var circuitBreaker = new CircuitBreaker.Net.CircuitBreaker(
  TaskScheduler.Default,
  maxFailures: 3,
  invocationTimeout: TimeSpan.FromSeconds(4),
  circuitResetTimeout: TimeSpan.FromSeconds(20));

// DownstreamServiceClient is a very small wrapper
// around HttpClient to add a couple of strongly-typed
// operations.
var client = new DownstreamServiceClient();
try
{
  // client.GetDataAsync() does async/await around
  // HttpClient.GetAsync() and JSON deserialization around
  // the model. No magic.
  return await circuitBreaker.Execute(() => client.GetDataAsync());
}
catch (CircuitBreakerOpenException)
{
  // The service is unavailable.
  return null;
}
catch (CircuitBreakerTimeoutException ex)
{
  // The circuit timed out.
  return null;
}
```

If I stick breakpoints on the `return null` lines in the code they aren't ever hit; I always get data returned (after however very long the wait is).

Am I missing something obvious? Do you have any async/await examples floating about?

**Comment by alexandrnikitin** [comment]
Hi Travis,

The library doesn't support async/await yet. I made some quick hacking in #2 You can take a look at the example: https://github.com/alexandrnikitin/CircuitBreaker.Net/pull/2/files#diff-0845b8c16762d4759c1865fae508a237R51

**Comment by alexandrnikitin** [comment]
I published `0.2.0-alpha` just to play with it. It is not covered with tests. Timeouts for async don't work. :blush: 
I will continue tomorrow or day after.

**Comment by tillig** [comment]
I look forward to it. Thanks!

**Comment by alexandrnikitin** [comment]
@tillig published [the 0.2.2 version](https://www.nuget.org/packages/CircuitBreaker.Net/0.2.2) with async/await support. Feel free to try it :smile:

**Comment by tillig** [comment]
Works _awesome_. Thanks! :sunglasses:
